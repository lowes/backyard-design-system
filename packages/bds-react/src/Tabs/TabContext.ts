import * as React from 'react'

const TabContext = React.createContext({
    selected: 0,
    // eslint-disable-next-line @typescript-eslint/no-unused-vars
    handleClick: (index: number, event: React.MouseEvent | React.KeyboardEvent) => {},
    // eslint-disable-next-line @typescript-eslint/no-unused-vars
    handleKeyDown: (index: number, event: React.KeyboardEvent) => {},
    isOnLayer: false
})

export default TabContext
