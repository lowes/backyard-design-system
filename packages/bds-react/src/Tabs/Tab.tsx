/* eslint-disable jsx-a11y/no-noninteractive-element-interactions */
import * as React from 'react'
import classNames from 'classnames'

import type { BackyardBaseProps } from '../utils/typings/BackyardProps'
import type { BDSForwardRef } from '../utils/typings/BDSComponentProps'

import { TabWrapper } from './style/TabWrapper'

import TabContext from './TabContext'
import { Typography } from '../Typography'

/**
 * Backyard React Tab Component
 *
 *
 */
const Tab: BDSForwardRef<TabProps> = React.forwardRef<TabRef, TabProps>(function Tab(
    {
        ariaControls,
        className,
        disabled = false,
        id,
        index,
        label = 'Label',
        onClick,
        onKeyDown,
        role = 'tab',
        ...props
    },
    ref,
) {
    // pull handleClick, handleKeyDown, and isOnLayer
    const { selected, handleClick, handleKeyDown, isOnLayer } = React.useContext(TabContext)

    /**
     * Coerces a boolean value to a string.
     *
     * @param value - the boolean value to coerce
     */
    const booleanCoersion = (value) => {
        if (value) return 'true'

        return 'false'
    }

    /**
     * Layout:
     *
     * <li>
     *   <a>
     *     <Typography />
     *   </a>
     * </li>
     */
    return (
        <TabWrapper
            className={classNames(
                `tab`,
                {
                    disabled,
                    selected: selected === index,
                },
                className,
            )}
            isOnLayer={isOnLayer}
            id={id}
            disabled={disabled}
            onClick={(event) => {
                if (!disabled) {
                    handleClick(index, event)

                    if (onClick) {
                        onClick(event)
                    }
                }
            }}
            onKeyDown={(event) => {
                if (!disabled) {
                    if (onKeyDown) {
                        onKeyDown(event)
                    }

                    if (handleKeyDown) {
                        handleKeyDown(index, event)
                    }
                }
            }}
            aria-controls={ariaControls}
            role={role}
            aria-selected={booleanCoersion(selected === index)}
            {...props}
            ref={ref}
        >
            <Typography variant="body_2" bold={selected === index}>
                {label}
            </Typography>
        </TabWrapper>
    )
})

type TabRef = HTMLButtonElement

interface TabProps extends BackyardBaseProps<TabRef> {
    /**
     * The id of the TabContent associated with the tab.
     */
    ariaControls?: string
    /**
     * Adds a class to the tab
     */
    className?: string
    /**
     * Wether or not the tab is disabled
     */
    disabled?: boolean
    /**
     * Adds an id to the tab
     */
    id?: string
    /**
     * The index of the tab
     */
    index?: number
    /**
     * The text within a tab.
     */
    label?: string
    /**
     * Trigger function for onClick.
     */
    onClick?: (event: React.MouseEvent) => void
    /**
     * Trigger function for onKeyDown.
     */
    onKeyDown?: (event: React.KeyboardEvent) => void
    /**
     * The role for the tab.
     */
    role?: 'tab' | 'presentation'
    /**
     * Wether or not the tab is selected
     */
    selected?: boolean
}

Tab.bdsName = 'Tab'

export type { TabRef, TabProps }

export { Tab }

export default Tab
