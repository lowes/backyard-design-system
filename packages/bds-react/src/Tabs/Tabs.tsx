import * as React from 'react'
import classNames from 'classnames'

import ChevronLeft from '@lowes-tech/bds-icons/ChevronLeft'
import ChevronRight from '@lowes-tech/bds-icons/ChevronRight'

import BaseTabsWrapper from './style/TabsWrapper'

import Carousel, { CarouselItem } from '../Carousel'
import type { BackyardBaseProps } from '../utils/typings/BackyardProps'
import type { BDSForwardRef } from '../utils/typings/BDSComponentProps'
import { useBackyardTheme, getShape } from '../ThemeProvider'

import TabContext from './TabContext'
import TabContent from './TabContent'
import type { TabRef } from './Tab'

/**
 * Backyard React Tabs Component
 */
const Tabs: BDSForwardRef<TabsProps> = React.forwardRef<TabsRef, TabsProps>(function Tabs(
    {
        children,
        className,
        defaultSelected = 0,
        shape: shapeProp, // = 'rounded',
        isOnLayer = false,
        keepMounted = false,
        handleOnChange,
        ...props
    },
    ref,
) {
    // Get Backyard Theme Context
    const theme = useBackyardTheme({
        validate: true,
        name: 'Tabs',
    })

    const { shape: shapeContext } = theme.context

    // Calculate shape
    const shape = getShape(shapeProp, shapeContext)

    // the index of the current selected tab
    const [selected, setSelected] = React.useState(defaultSelected)

    // ref of tab list
    const tabListRef = React.useRef<HTMLDivElement>(null)

    /**
     * The `handleClick` prop.
     *
     * This is passed to the tab component's `onClick` property and is called inside
     * of the Tab component.
     *
     * @param index - the index of the tab within the list of the tabs
     */
    const handleClick = (index: number, event: React.MouseEvent) => {
        setSelected(index)

        if (handleOnChange) {
            handleOnChange(event)
        }
    } // (event: React.KeyboardEvent | React.MouseEvent) => void

    /**
     * Get enabled sibling through recursion
     *
     * @param direction - direction to recursively check
     * @param element - current tab
     */
    const getEnabledSibling = (direction: 'previous' | 'next', element: TabRef) => {
        // Get next or previous sibling depending on dirction
        const sibling = (direction === 'next'
            ? element.nextElementSibling
            : element.previousElementSibling) as TabRef

        // If sibling found,
        if (sibling) {
            // If sibling is disabled,
            if (sibling.disabled) {
                // Try next sibling
                return getEnabledSibling(direction, sibling as TabRef)
            }

            // Else, sibling is not disabled
            // Return enabled sibling
            return sibling
        }

        // Else, sibling was not found,
        // Return no sibling
        return null
    }

    /**
     * The `handleKeyDown` trigger.
     *
     * This is called inside of the Tab component.
     *
     * @param id - the id of the tab
     * @param index - the index of the tab within the list of tabs
     */
    const handleKeyDown = (index: number, event: React.KeyboardEvent<TabRef>) => {
        // get tab from event target
        const tab = event.target as TabRef

        // default target to null
        let target: TabRef = null

        // where key down...
        switch (event.key) {
            // is left arrow...
            case 'ArrowLeft':
                // target is previous tab (if available)
                target = getEnabledSibling('previous', tab) as TabRef
                break

            case 'ArrowRight':
                // target is next tab (if available)
                target = getEnabledSibling('next', tab) as TabRef
                break

            default:
                // Do nothing
                break
        }

        if (target !== null) {
            target.focus()
        }

        if (event.key === 'Enter' || event.key === ' ') {
            setSelected(index)
        }

        if (handleOnChange) {
            handleOnChange(event)
        }
    }

    /**
     * Returns the tabs with event handlers and their respective index passed through.
     */
    const getTabs = (role: string) =>
        React.Children.map(children, (tab, index) =>
            // @ts-ignore
            React.cloneElement(tab, {
                // @ts-ignore
                ariaControls: `${tab.props.id}-panel`,
                role,
                index,
            }),
        )

    // The tabs mapped with the event handlers and their respective index
    const tabs = theme.isMobile ? getTabs('') : getTabs('tab')

    // if is mobile, wrap the tabs in a carousel
    // else return the tabs
    const computedTabs = theme.isMobile ? (
        <Carousel
            id="mobile-tabs-scrollbar"
            carouselButtonProps={{
                shape: 'squared',
                style: {
                    border: 'none',
                },
            }}
            role="tablist"
            leftIcon={<ChevronLeft />}
            noScrollbar
            rightIcon={<ChevronRight />}
            ref={tabListRef}
        >
            {tabs.map((tab, index) => (
                <CarouselItem key={index}>{tab}</CarouselItem>
            ))}
        </Carousel>
    ) : (
        <div className="tablist" role="tablist" ref={tabListRef}>
            {tabs}
        </div>
    )

    /**
     * Layout:
     * <div tabs role='navigation'>
     *   <
     * </div>
     */
    return (
        <TabContext.Provider
            value={{
                selected,
                handleClick,
                handleKeyDown,
                isOnLayer,
            }}
        >
            <BaseTabsWrapper
                className={classNames('tabs-wrapper', `shape--${shape}`, className)}
                isOnLayer={isOnLayer}
                role="navigation"
                {...props}
                ref={ref}
            >
                <div className="tabs">{computedTabs}</div>
                {
                    /**
                     * If keepMounted is true, we display only the selected tab's content.
                     *
                     * If keepMounted is false, we display all tab content, but only set
                     * the selected tab's content to be visible. The keepMounted prop should
                     * be set to false when content will have an impact on SEO.
                     */
                    !keepMounted ? (
                        <TabContent
                            id={`${
                                tabs.find((tab) => tab.props.index === selected).props.id
                            }-panel`}
                            selected
                        >
                            {tabs.find((tab) => tab.props.index === selected).props.children}
                        </TabContent>
                    ) : (
                        React.Children.map(tabs, (tab, index) => {
                            return (
                                <TabContent
                                    key={index}
                                    id={`${tab.props.id}-panel`}
                                    selected={selected === index}
                                >
                                    {tab.props.children}
                                </TabContent>
                            )
                        })
                    )
                }
            </BaseTabsWrapper>
        </TabContext.Provider>
    )
})

type TabsRef = HTMLDivElement

interface TabsProps extends BackyardBaseProps<TabsRef> {
    /**
     * The Tab components.
     */
    children?: React.ReactElement[]
    /**
     * Adds a class name to the Tab component's wrapper.
     */
    className?: string
    /**
     * Shape of the edges of the tabs
     */
    shape?: 'rounded' | 'squared'
    /**
     * Iverses the field color token. This should be used when the Tabs component is on a layer color.
     */
    isOnLayer?: boolean
    /**
     * If false, the TabContent for each tab is not attached to the DOM.
     *
     * If true, the TabContent for each tab is attached to the DOM. This
     * option is best if you need the content in the DOM for SEO purposes.
     */
    keepMounted?: boolean
    /**
     * Sends a mouse or keyboard event based on how the change occurred.
     */
    handleOnChange?: (event: React.MouseEvent | React.KeyboardEvent) => void
    /**
     * The tab that is selected by default.
     */
    defaultSelected?: number
}

Tabs.bdsName = 'Tabs'

export type { TabsRef, TabsProps }

export { Tabs }

export default Tabs
