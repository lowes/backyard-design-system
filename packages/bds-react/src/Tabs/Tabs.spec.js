import * as React from 'react'
import { render } from '../../test-utils'
import { ThemeProvider } from '../ThemeProvider'
import Tabs from './Tabs'
import Tab from './Tab'
import Typography from '../Typography'

describe('Tabs Snapshots', () => {
    test('Tabs default snapshot', () => {
        const { asFragment } = render(
            <Tabs>
                <Tab id='tab-1' label='Tab one'>
                    <Typography>This is tab one's content.</Typography>
                </Tab>
            </Tabs>
        )

        expect(asFragment()).toMatchSnapshot()
    })

    test('Tabs in different themes', () => {
        const themes = ['light', 'dark']

        themes.forEach(theme => {
            const { asFragment } = render(
                <Tabs>
                    <Tab id='tab-1' label='Tab one'>
                        <Typography>This is tab one's content</Typography>
                    </Tab>
                    <Tab id='tab-2' label='Tab two'>
                        <Typography>This is tab two's content</Typography>
                    </Tab>
                    <Tab id='tab-3' label='Tab three'>
                        <Typography>This is tab three's content</Typography>
                    </Tab>
                </Tabs>
            )

            expect(asFragment()).toMatchSnapshot()
        })
    })
})