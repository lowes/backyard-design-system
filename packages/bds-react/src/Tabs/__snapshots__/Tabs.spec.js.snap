// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs Snapshots Tabs default snapshot 1`] = `
<DocumentFragment>
  .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
}

.c0.shape--rounded .tablist > .tab {
  border-top-left-radius: 0;
  border-top-right-radius: 0;
  border-bottom-left-radius: 0;
  border-bottom-right-radius: 0;
}

.c0.shape--squared .tablist {
  border-radius: 0;
}

.c0.shape--squared .tablist > .tab {
  border-top-left-radius: 0;
  border-top-right-radius: 0;
  border-bottom-left-radius: 0;
  border-bottom-right-radius: 0;
}

.c0 .tabs {
  position: relative;
  width: 100%;
  z-index: 1;
  padding: 0;
}

.c0 .tabs .tablist {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  width: 100%;
  background-color: trasparent;
  margin: 0;
  padding: 0;
  list-style: none;
}

.c0 .content-container {
  width: 100%;
}

.c0 .tabs .carousel-item {
  margin: 0;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c0 .tabs .carousel-button-left.carousel-button-left.carousel-button-left.carousel-button-left,
.c0 .tabs .carousel-button-right.carousel-button-right.carousel-button-right.carousel-button-right {
  background-color: transparent;
  width: 2.375rem;
  height: 2.375rem;
  padding: auto;
}

.c0 .tabs .carousel-button-left.carousel-button-left.carousel-button-left.carousel-button-left:hover,
.c0 .tabs .carousel-button-right.carousel-button-right.carousel-button-right.carousel-button-right:hover {
  border-bottom-color: var(--bds-color-action-interactive-hover);
}

.c0 .tabs .carousel-button-left.carousel-button-left.carousel-button-left.carousel-button-left:focus,
.c0 .tabs .carousel-button-right.carousel-button-right.carousel-button-right.carousel-button-right:focus {
  box-shadow: inset 0 0 0 var(--bds-sizes-size-2) var(--bds-color-action-interactive);
}

.c0 .tabs .carousel-button-left.carousel-button-left.carousel-button-left.carousel-button-left .icon path,
.c0 .tabs .carousel-button-right.carousel-button-right.carousel-button-right.carousel-button-right .icon path {
  fill: hsl(213,84%,45%,1);
}

.c0 .tabs .carousel-content {
  width: 100%;
  padding: 0;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
}

.c3 {
  display: none;
  padding: 0;
}

.c3.selected {
  display: block;
}

.c1.tab {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  white-space: nowrap;
  width: 100%;
  height: var(--bds-sizes-size-40);
  outline: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  padding: 0 var(--bds-sizes-size-20);
  margin: 0;
  border-radius: 0;
  border: none;
  border-bottom: 1px solid var(--bds-color-border-subdued);
  background-color: transparent;
  cursor: pointer;
}

.c1.tab:hover:not(.selected):not(.disabled),
.c1.tab.hover:not(.selected):not(.disabled) {
  border-bottom-color: var(--bds-color-action-interactive);
}

.c1.tab.selected {
  border-bottom-color: var(--bds-color-action-interactive);
  box-shadow: 0 1px 0 var(--bds-color-action-interactive);
}

.c1.tab.selected:focus,
.c1.tab.selected.focus {
  outline: none;
  z-index: 1;
  box-shadow: 0 0 0 var(--bds-sizes-size-1) var(--bds-color-action-interactive);
}

.c1.tab.disabled {
  z-index: 0;
  cursor: not-allowed;
}

.c1.tab.disabled > p {
  color: var(--bds-color-text-disabled);
}

.c1.tab:focus,
.c1.tab.focus {
  outline: none;
  z-index: 1;
  box-shadow: 0 0 0 var(--bds-sizes-size-1) var(--bds-color-action-interactive);
}

.c2 {
  color: hsl(0,0%,0%,.87);
  margin: 0;
  font-family: Roboto,Helvetica,Arial,sans-serif;
  font-weight: var(--bds-font-weight-regular);
  color: hsl(0,0%,0%,.87);
}

.c2.align--left {
  text-align: left;
}

.c2.align--center {
  text-align: center;
}

.c2.align--right {
  text-align: right;
}

.c2.display--block {
  display: block;
}

.c2.display--inline {
  display: inline;
}

.c2 b,
.c2 bold,
.c2 strong {
  font-family: Roboto,Helvetica,Arial,sans-serif;
  font-weight: 500;
}

.c2.bold {
  font-weight: 500;
}

.c2.c2.c2.body_1 {
  font-size: var(--bds-font-tag-body-1-mobile-font-size);
  line-height: var(--bds-font-tag-body-1-mobile-line-height);
}

.c2.c2.c2.body_2 {
  font-size: var(--bds-font-tag-body-2-mobile-font-size);
  line-height: var(--bds-font-tag-body-2-mobile-line-height);
}

@media only screen and (min-width:var(--bds-grid-breakpoint-md-max)) {
  .c3 {
    padding-top: var(--bds-sizes-size-16);
  }
}

@media only screen and (min-width:66.0625rem) {
  .c2.c2.c2.body_1 {
    font-size: var(--bds-font-tag-body-1-desktop-font-size);
    line-height: var(--bds-font-tag-body-1-desktop-line-height);
  }

  .c2.c2.c2.body_2 {
    font-size: var(--bds-font-tag-body-2-desktop-font-size);
    line-height: var(--bds-font-tag-body-2-desktop-line-height);
  }
}

<div
    class="c0 tabs-wrapper shape--rounded"
    role="navigation"
  >
    <div
      class="tabs"
    >
      <div
        class="tablist"
        role="tablist"
      >
        <button
          aria-controls="tab-1-panel"
          aria-selected="true"
          class="c1 tab selected"
          id="tab-1"
          role="tab"
        >
          <p
            class="c2 typography variant--body_2 align--left bold body_2"
            color="text_primary"
          >
            Tab one
          </p>
        </button>
      </div>
    </div>
    <div
      class="c3 tab-content selected"
      id="tab-1-panel"
      role="tabpanel"
    >
      <span
        class="typography variant--body align--left"
        color="text_primary"
      >
        This is tab one's content.
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Tabs Snapshots Tabs in different themes 1`] = `
<DocumentFragment>
  .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
}

.c0.shape--rounded .tablist > .tab {
  border-top-left-radius: 0;
  border-top-right-radius: 0;
  border-bottom-left-radius: 0;
  border-bottom-right-radius: 0;
}

.c0.shape--squared .tablist {
  border-radius: 0;
}

.c0.shape--squared .tablist > .tab {
  border-top-left-radius: 0;
  border-top-right-radius: 0;
  border-bottom-left-radius: 0;
  border-bottom-right-radius: 0;
}

.c0 .tabs {
  position: relative;
  width: 100%;
  z-index: 1;
  padding: 0;
}

.c0 .tabs .tablist {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  width: 100%;
  background-color: trasparent;
  margin: 0;
  padding: 0;
  list-style: none;
}

.c0 .content-container {
  width: 100%;
}

.c0 .tabs .carousel-item {
  margin: 0;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c0 .tabs .carousel-button-left.carousel-button-left.carousel-button-left.carousel-button-left,
.c0 .tabs .carousel-button-right.carousel-button-right.carousel-button-right.carousel-button-right {
  background-color: transparent;
  width: 2.375rem;
  height: 2.375rem;
  padding: auto;
}

.c0 .tabs .carousel-button-left.carousel-button-left.carousel-button-left.carousel-button-left:hover,
.c0 .tabs .carousel-button-right.carousel-button-right.carousel-button-right.carousel-button-right:hover {
  border-bottom-color: var(--bds-color-action-interactive-hover);
}

.c0 .tabs .carousel-button-left.carousel-button-left.carousel-button-left.carousel-button-left:focus,
.c0 .tabs .carousel-button-right.carousel-button-right.carousel-button-right.carousel-button-right:focus {
  box-shadow: inset 0 0 0 var(--bds-sizes-size-2) var(--bds-color-action-interactive);
}

.c0 .tabs .carousel-button-left.carousel-button-left.carousel-button-left.carousel-button-left .icon path,
.c0 .tabs .carousel-button-right.carousel-button-right.carousel-button-right.carousel-button-right .icon path {
  fill: hsl(213,84%,45%,1);
}

.c0 .tabs .carousel-content {
  width: 100%;
  padding: 0;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
}

.c3 {
  display: none;
  padding: 0;
}

.c3.selected {
  display: block;
}

.c1.tab {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  white-space: nowrap;
  width: 100%;
  height: var(--bds-sizes-size-40);
  outline: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  padding: 0 var(--bds-sizes-size-20);
  margin: 0;
  border-radius: 0;
  border: none;
  border-bottom: 1px solid var(--bds-color-border-subdued);
  background-color: transparent;
  cursor: pointer;
}

.c1.tab:hover:not(.selected):not(.disabled),
.c1.tab.hover:not(.selected):not(.disabled) {
  border-bottom-color: var(--bds-color-action-interactive);
}

.c1.tab.selected {
  border-bottom-color: var(--bds-color-action-interactive);
  box-shadow: 0 1px 0 var(--bds-color-action-interactive);
}

.c1.tab.selected:focus,
.c1.tab.selected.focus {
  outline: none;
  z-index: 1;
  box-shadow: 0 0 0 var(--bds-sizes-size-1) var(--bds-color-action-interactive);
}

.c1.tab.disabled {
  z-index: 0;
  cursor: not-allowed;
}

.c1.tab.disabled > p {
  color: var(--bds-color-text-disabled);
}

.c1.tab:focus,
.c1.tab.focus {
  outline: none;
  z-index: 1;
  box-shadow: 0 0 0 var(--bds-sizes-size-1) var(--bds-color-action-interactive);
}

.c2 {
  color: hsl(0,0%,0%,.87);
  margin: 0;
  font-family: Roboto,Helvetica,Arial,sans-serif;
  font-weight: var(--bds-font-weight-regular);
  color: hsl(0,0%,0%,.87);
}

.c2.align--left {
  text-align: left;
}

.c2.align--center {
  text-align: center;
}

.c2.align--right {
  text-align: right;
}

.c2.display--block {
  display: block;
}

.c2.display--inline {
  display: inline;
}

.c2 b,
.c2 bold,
.c2 strong {
  font-family: Roboto,Helvetica,Arial,sans-serif;
  font-weight: 500;
}

.c2.bold {
  font-weight: 500;
}

.c2.c2.c2.body_1 {
  font-size: var(--bds-font-tag-body-1-mobile-font-size);
  line-height: var(--bds-font-tag-body-1-mobile-line-height);
}

.c2.c2.c2.body_2 {
  font-size: var(--bds-font-tag-body-2-mobile-font-size);
  line-height: var(--bds-font-tag-body-2-mobile-line-height);
}

@media only screen and (min-width:var(--bds-grid-breakpoint-md-max)) {
  .c3 {
    padding-top: var(--bds-sizes-size-16);
  }
}

@media only screen and (min-width:66.0625rem) {
  .c2.c2.c2.body_1 {
    font-size: var(--bds-font-tag-body-1-desktop-font-size);
    line-height: var(--bds-font-tag-body-1-desktop-line-height);
  }

  .c2.c2.c2.body_2 {
    font-size: var(--bds-font-tag-body-2-desktop-font-size);
    line-height: var(--bds-font-tag-body-2-desktop-line-height);
  }
}

<div
    class="c0 tabs-wrapper shape--rounded"
    role="navigation"
  >
    <div
      class="tabs"
    >
      <div
        class="tablist"
        role="tablist"
      >
        <button
          aria-controls="tab-1-panel"
          aria-selected="true"
          class="c1 tab selected"
          id="tab-1"
          role="tab"
        >
          <p
            class="c2 typography variant--body_2 align--left bold body_2"
            color="text_primary"
          >
            Tab one
          </p>
        </button>
        <button
          aria-controls="tab-2-panel"
          aria-selected="false"
          class="c1 tab"
          id="tab-2"
          role="tab"
        >
          <p
            class="c2 typography variant--body_2 align--left body_2"
            color="text_primary"
          >
            Tab two
          </p>
        </button>
        <button
          aria-controls="tab-3-panel"
          aria-selected="false"
          class="c1 tab"
          id="tab-3"
          role="tab"
        >
          <p
            class="c2 typography variant--body_2 align--left body_2"
            color="text_primary"
          >
            Tab three
          </p>
        </button>
      </div>
    </div>
    <div
      class="c3 tab-content selected"
      id="tab-1-panel"
      role="tabpanel"
    >
      <span
        class="typography variant--body align--left"
        color="text_primary"
      >
        This is tab one's content
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Tabs Snapshots Tabs in different themes 2`] = `
<DocumentFragment>
  .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
}

.c0.shape--rounded .tablist > .tab {
  border-top-left-radius: 0;
  border-top-right-radius: 0;
  border-bottom-left-radius: 0;
  border-bottom-right-radius: 0;
}

.c0.shape--squared .tablist {
  border-radius: 0;
}

.c0.shape--squared .tablist > .tab {
  border-top-left-radius: 0;
  border-top-right-radius: 0;
  border-bottom-left-radius: 0;
  border-bottom-right-radius: 0;
}

.c0 .tabs {
  position: relative;
  width: 100%;
  z-index: 1;
  padding: 0;
}

.c0 .tabs .tablist {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  width: 100%;
  background-color: trasparent;
  margin: 0;
  padding: 0;
  list-style: none;
}

.c0 .content-container {
  width: 100%;
}

.c0 .tabs .carousel-item {
  margin: 0;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c0 .tabs .carousel-button-left.carousel-button-left.carousel-button-left.carousel-button-left,
.c0 .tabs .carousel-button-right.carousel-button-right.carousel-button-right.carousel-button-right {
  background-color: transparent;
  width: 2.375rem;
  height: 2.375rem;
  padding: auto;
}

.c0 .tabs .carousel-button-left.carousel-button-left.carousel-button-left.carousel-button-left:hover,
.c0 .tabs .carousel-button-right.carousel-button-right.carousel-button-right.carousel-button-right:hover {
  border-bottom-color: var(--bds-color-action-interactive-hover);
}

.c0 .tabs .carousel-button-left.carousel-button-left.carousel-button-left.carousel-button-left:focus,
.c0 .tabs .carousel-button-right.carousel-button-right.carousel-button-right.carousel-button-right:focus {
  box-shadow: inset 0 0 0 var(--bds-sizes-size-2) var(--bds-color-action-interactive);
}

.c0 .tabs .carousel-button-left.carousel-button-left.carousel-button-left.carousel-button-left .icon path,
.c0 .tabs .carousel-button-right.carousel-button-right.carousel-button-right.carousel-button-right .icon path {
  fill: hsl(213,84%,45%,1);
}

.c0 .tabs .carousel-content {
  width: 100%;
  padding: 0;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
}

.c3 {
  display: none;
  padding: 0;
}

.c3.selected {
  display: block;
}

.c1.tab {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  white-space: nowrap;
  width: 100%;
  height: var(--bds-sizes-size-40);
  outline: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  padding: 0 var(--bds-sizes-size-20);
  margin: 0;
  border-radius: 0;
  border: none;
  border-bottom: 1px solid var(--bds-color-border-subdued);
  background-color: transparent;
  cursor: pointer;
}

.c1.tab:hover:not(.selected):not(.disabled),
.c1.tab.hover:not(.selected):not(.disabled) {
  border-bottom-color: var(--bds-color-action-interactive);
}

.c1.tab.selected {
  border-bottom-color: var(--bds-color-action-interactive);
  box-shadow: 0 1px 0 var(--bds-color-action-interactive);
}

.c1.tab.selected:focus,
.c1.tab.selected.focus {
  outline: none;
  z-index: 1;
  box-shadow: 0 0 0 var(--bds-sizes-size-1) var(--bds-color-action-interactive);
}

.c1.tab.disabled {
  z-index: 0;
  cursor: not-allowed;
}

.c1.tab.disabled > p {
  color: var(--bds-color-text-disabled);
}

.c1.tab:focus,
.c1.tab.focus {
  outline: none;
  z-index: 1;
  box-shadow: 0 0 0 var(--bds-sizes-size-1) var(--bds-color-action-interactive);
}

.c2 {
  color: hsl(0,0%,0%,.87);
  margin: 0;
  font-family: Roboto,Helvetica,Arial,sans-serif;
  font-weight: var(--bds-font-weight-regular);
  color: hsl(0,0%,0%,.87);
}

.c2.align--left {
  text-align: left;
}

.c2.align--center {
  text-align: center;
}

.c2.align--right {
  text-align: right;
}

.c2.display--block {
  display: block;
}

.c2.display--inline {
  display: inline;
}

.c2 b,
.c2 bold,
.c2 strong {
  font-family: Roboto,Helvetica,Arial,sans-serif;
  font-weight: 500;
}

.c2.bold {
  font-weight: 500;
}

.c2.c2.c2.body_1 {
  font-size: var(--bds-font-tag-body-1-mobile-font-size);
  line-height: var(--bds-font-tag-body-1-mobile-line-height);
}

.c2.c2.c2.body_2 {
  font-size: var(--bds-font-tag-body-2-mobile-font-size);
  line-height: var(--bds-font-tag-body-2-mobile-line-height);
}

@media only screen and (min-width:var(--bds-grid-breakpoint-md-max)) {
  .c3 {
    padding-top: var(--bds-sizes-size-16);
  }
}

@media only screen and (min-width:66.0625rem) {
  .c2.c2.c2.body_1 {
    font-size: var(--bds-font-tag-body-1-desktop-font-size);
    line-height: var(--bds-font-tag-body-1-desktop-line-height);
  }

  .c2.c2.c2.body_2 {
    font-size: var(--bds-font-tag-body-2-desktop-font-size);
    line-height: var(--bds-font-tag-body-2-desktop-line-height);
  }
}

<div
    class="c0 tabs-wrapper shape--rounded"
    role="navigation"
  >
    <div
      class="tabs"
    >
      <div
        class="tablist"
        role="tablist"
      >
        <button
          aria-controls="tab-1-panel"
          aria-selected="true"
          class="c1 tab selected"
          id="tab-1"
          role="tab"
        >
          <p
            class="c2 typography variant--body_2 align--left bold body_2"
            color="text_primary"
          >
            Tab one
          </p>
        </button>
        <button
          aria-controls="tab-2-panel"
          aria-selected="false"
          class="c1 tab"
          id="tab-2"
          role="tab"
        >
          <p
            class="c2 typography variant--body_2 align--left body_2"
            color="text_primary"
          >
            Tab two
          </p>
        </button>
        <button
          aria-controls="tab-3-panel"
          aria-selected="false"
          class="c1 tab"
          id="tab-3"
          role="tab"
        >
          <p
            class="c2 typography variant--body_2 align--left body_2"
            color="text_primary"
          >
            Tab three
          </p>
        </button>
      </div>
    </div>
    <div
      class="c3 tab-content selected"
      id="tab-1-panel"
      role="tabpanel"
    >
      <span
        class="typography variant--body align--left"
        color="text_primary"
      >
        This is tab one's content
      </span>
    </div>
  </div>
</DocumentFragment>
`;
